src/paint/map_element/scenery.c

```diff
@@ -16,8 +16,309 @@
 
 #include "map_element.h"
 #include "../../addresses.h"
+#include "../../config.h"
+#include "../../game.h"
+#include "../../interface/viewport.h"
+#include "../../paint/paint.h"
+#include "../../paint/supports.h"
 #include "../../world/map.h"
+#include "../../world/scenery.h"
 
 void scenery_paint(uint8 direction, int height, rct_map_element* mapElement) {
-	RCT2_CALLPROC_X(0x6DFF47, 0, 0, direction, height, (int)mapElement, 0, 0);
+	//RCT2_CALLPROC_X(0x6DFF47, 0, 0, direction, height, (int)mapElement, 0, 0); return;
+	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8_t) = VIEWPORT_INTERACTION_ITEM_SCENERY;
+	rct_xyz16 boxlength;
+	rct_xyz16 boxoffset;
+	boxoffset.x = 0;
+	boxoffset.y = 0;
+	boxoffset.z = height;
+	int baseImageid = 0;
+	if (RCT2_GLOBAL(0x009DEA6F, uint8) & 1) {
+		rct_map_element *mapElement2 = RCT2_GLOBAL(0x00F63674, rct_map_element*);
```

Comment:
This is a check to see if the map element is part of the track design selection. Use `track_design_save_contains_map_element`